servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/SupportedBufferAllocatorsTest.java

```diff
@@ -0,0 +1,86 @@
+import org.junit.runners.Parameterized;
+
+import java.util.Collection;
+
+import static io.servicetalk.buffer.api.ReadOnlyBufferAllocators.PREFER_DIRECT_RO_ALLOCATOR;
+import static io.servicetalk.buffer.api.ReadOnlyBufferAllocators.PREFER_HEAP_RO_ALLOCATOR;
+import static io.servicetalk.buffer.netty.BufferAllocators.PREFER_DIRECT_ALLOCATOR;
+import static io.servicetalk.buffer.netty.BufferAllocators.PREFER_HEAP_ALLOCATOR;
+import static io.servicetalk.concurrent.api.Publisher.from;
+import static io.servicetalk.http.api.HttpApiConversions.toStreamingHttpService;
+import static io.servicetalk.http.api.HttpResponseStatus.OK;
+import static io.servicetalk.http.netty.AbstractNettyHttpServerTest.ExecutorSupplier.CACHED;
+import static io.servicetalk.http.netty.AbstractNettyHttpServerTest.ExecutorSupplier.CACHED_SERVER;
+import static io.servicetalk.http.netty.HttpProtocol.HTTP_1;
+import static io.servicetalk.http.netty.HttpProtocol.HTTP_2;
+import static java.nio.charset.StandardCharsets.US_ASCII;
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+
+@RunWith(Parameterized.class)
+public class SupportedBufferAllocatorsTest extends AbstractNettyHttpServerTest {
+
+    private final HttpProtocol protocol;
+    private final BufferAllocator allocator;
+
+    public SupportedBufferAllocatorsTest(HttpProtocol protocol, BufferAllocator allocator) {
+        super(CACHED, CACHED_SERVER);
+        this.protocol = protocol;
+        this.allocator = allocator;
+        protocol(protocol.config);
+    }
+
+    @Parameterized.Parameters(name = "{index}: protocol={0}, allocator={1}")
+    public static Collection<Object[]> data() {
+        return asList(
+                new Object[]{HTTP_1, PREFER_HEAP_ALLOCATOR},
+                new Object[]{HTTP_1, PREFER_DIRECT_ALLOCATOR},
+                new Object[]{HTTP_1, PREFER_HEAP_RO_ALLOCATOR},
+                new Object[]{HTTP_1, PREFER_DIRECT_RO_ALLOCATOR},
+                new Object[]{HTTP_2, PREFER_HEAP_ALLOCATOR},
+                new Object[]{HTTP_2, PREFER_DIRECT_ALLOCATOR},
+                new Object[]{HTTP_2, PREFER_HEAP_RO_ALLOCATOR},
+                new Object[]{HTTP_2, PREFER_DIRECT_RO_ALLOCATOR});
+    }
+
+    @Override
+    protected void service(final StreamingHttpService service) {
+        super.service((toStreamingHttpService((BlockingHttpService) (ctx, request, responseFactory) ->
+                        responseFactory.ok().payloadBody(allocator.fromAscii(request.payloadBody().toString(US_ASCII))),
+                strategy -> strategy)).adaptor());
+    }
+
+    @Test
+    public void test() throws Exception {
+        String payload = "Hello ServiceTalk";
+        StreamingHttpRequest request = streamingHttpConnection().post("/")
+                .payloadBody(from(allocator.fromAscii(payload)));
+        StreamingHttpResponse response = makeRequest(request);
+        assertResponse(response, protocol.version, OK, singletonList(payload));
```

Comment:
not introduced by this PR but the validation of the `getBodyAsListOfStrings` doesn't seem right. The way data is written may not necessarily be the way data is read. fixed in https://github.com/apple/servicetalk/pull/1173